packages/core/src/utils/StorageUtil.ts

```diff
@@ -5,19 +5,22 @@ import { ChainController } from '../controllers/ChainController.js'
 
 // -- Utility -----------------------------------------------------------------
 export const StorageUtil = {
-  setWalletConnectDeepLink({ href }: { href: string; name: string }) {
+  setWalletConnectDeepLink({ name, href }: { href: string; name: string }) {
     try {
-      SafeLocalStorage.setItem(SafeLocalStorageKeys.DEEPLINK_CHOICE, href)
+      const deeplink = JSON.stringify({ href, name })
+      SafeLocalStorage.setItem(SafeLocalStorageKeys.DEEPLINK, deeplink)
+      // Required by @walletconnect/universal-provider to handle deep links
+      localStorage.setItem('WALLETCONNECT_DEEPLINK_CHOICE', deeplink)
```

Comment:
This needs to stay as-is so that the UP sign client can fetch it. 

Still, should use `SafeLocalStorage`